src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmServiceImpl.java

```diff
@@ -101,15 +115,117 @@ public List<String> listBlobs(BlobContainerClient blobContainerClient, String pr
         return files;
     }
 
-    public PagedIterable<BlobItem> listBlobsHierarchicalListing(BlobContainerClient blobContainerClient,
-                                                                String delimiter,
-                                                                String prefix) {
+    private PagedIterable<BlobItem> listBlobsHierarchicalListing(BlobContainerClient blobContainerClient,
+                                                                 String delimiter,
+                                                                 String prefix) {
 
         ListBlobsOptions options = new ListBlobsOptions().setPrefix(prefix);
         Duration timeout = Duration.of(TIMEOUT, ChronoUnit.SECONDS);
         return blobContainerClient.listBlobsByHierarchy(delimiter, options, timeout);
     }
 
+    public List<String> listSubmissionBlobsUsingBatch(String containerName, String filename, Integer batchSize) {
+        BlobContainerClient containerClient = armDataManagementDao.getBlobContainerClient(containerName);
+        String prefix = armDataManagementConfiguration.getFolders().getSubmission() + filename;
+        return listBlobsUsingBatch(containerClient, prefix, batchSize);
+    }
+
+    public List<String> listResponseBlobsUsingBatch(String containerName, String filename, Integer batchSize) {
+        BlobContainerClient containerClient = armDataManagementDao.getBlobContainerClient(containerName);
+        String prefix = armDataManagementConfiguration.getFolders().getResponse() + filename;
+        return listBlobsUsingBatch(containerClient, prefix, batchSize);
+    }
+
+    private List<String> listBlobsUsingBatch(BlobContainerClient blobContainerClient, String blobPathAndName, Integer batchSize) {
+        List<String> files = new ArrayList<>();
+        log.debug("About to list files for {} with batch size {}", blobPathAndName, batchSize);
+        ListBlobsOptions options = new ListBlobsOptions()
+            .setPrefix(blobPathAndName)
+            .setMaxResultsPerPage(batchSize);
+        Duration timeout = Duration.of(TIMEOUT, ChronoUnit.SECONDS);
+
+        int pageNumber = 0;
+        Iterable<PagedResponse<BlobItem>> blobPages = blobContainerClient.listBlobs(options, timeout).iterableByPage();
+        for (PagedResponse<BlobItem> page : blobPages) {
```

Comment:
can't work out what this bit is doing or if pageNumber is ever anything other than 1 ! maybe it's just me :-/

Reply:
It was an attempt to batch using setMaxResultsPerPage but what that seems to do is just put results in pages so you only grab the first page.